x-pack/plugins/security_solution/public/management/pages/policy/view/artifacts/empty/use_policy_artifacts_empty_hooks.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  ENDPOINT_EVENT_FILTERS_LIST_ID,
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+} from '@kbn/securitysolution-list-constants';
+import { useNavigateToAppEventHandler } from '../../../../../../common/hooks/endpoint/use_navigate_to_app_event_handler';
+import { useAppUrl } from '../../../../../../common/lib/kibana/hooks';
+import {
+  getPolicyEventFiltersPath,
+  getEventFiltersListPath,
+  getTrustedAppsListPath,
+  getHostIsolationExceptionsListPath,
+  getPolicyDetailsArtifactsListPath,
+  getPolicyHostIsolationExceptionsPath,
+} from '../../../../../common/routing';
+import { APP_UI_ID } from '../../../../../../../common/constants';
+
+export const useGetLinkTo = (
+  policyId: string,
+  policyName: string,
+  listId: string,
+  location?: Partial<{ show: 'create' }>
+) => {
+  const { getAppUrl } = useAppUrl();
+  const { toRoutePath, toRouteUrl } = useMemo(() => {
+    let path = '';
+    if (listId === ENDPOINT_TRUSTED_APPS_LIST_ID) {
+      path = getTrustedAppsListPath(location);
+    } else if (listId === ENDPOINT_EVENT_FILTERS_LIST_ID) {
+      path = getEventFiltersListPath(location);
+    } else {
+      path = getHostIsolationExceptionsListPath(location);
+    }
+    return {
+      toRoutePath: path,
+      toRouteUrl: getAppUrl({ path }),
+    };
+  }, [getAppUrl, listId, location]);
```

Comment:
🤔 Is there a way we can remove this logic from here and require it on input to the artifact component? it would avoid us having to update this logic every time we introduce a new artifact

Reply:
Yeah, I was thinking that too, I will try to move this out